src/style/__tests__/scheme.spec.js

```diff
@@ -0,0 +1,94 @@
+let Scheme;
+describe('Scheme', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    Scheme = require('../scheme').default;
+  });
+
+  describe('initial values', () => {
+    it('should initial schemes be empty objects', () => {
+      expect(Scheme.schemes).toEqual({dark: {}, light: {}});
+    });
+
+    it('should initial current scheme type be "default"', () => {
+      expect(Scheme.currentScheme).toBe('default');
+    });
+
+    it('should retrieve actual scheme type (defaulting to "light")', () => {
+      expect(Scheme.getSchemeType()).toBe('light');
+    });
+  });
+
+  describe('setScheme', () => {
+    it('should initially be light', () => {
+      expect(Scheme.getSchemeType()).toBe('light');
+    });
+    it('should override default scheme', () => {
+      Scheme.setScheme('dark');
+      expect(Scheme.getSchemeType()).toBe('dark');
+    });
+
+    it('should throw on invalid scheme color', () => {
+      expect(() => Scheme.setScheme('yellow')).toThrow(`yellow is invalid colorScheme, please use 'light' | 'dark' | 'default'`);
+    });
+  });
+
+  describe('getScheme/loadScheme', () => {
+    beforeEach(() => {
+      Scheme.loadSchemes({
+        dark: {
+          a: 'black',
+          b: 'black'
+        },
+        light: {
+          a: 'white',
+          b: 'white'
+        }
+      });
+    });
+
+    it('should return scheme by current scheme type (initially by light)', () => {
+      expect(Scheme.getScheme()).toEqual({
+        a: 'white',
+        b: 'white'
+      });
+    });
+
+    it('should return correct scheme after setScheme has changed scheme type', () => {
+      Scheme.setScheme('dark');
+      expect(Scheme.getScheme()).toEqual({
+        a: 'black',
+        b: 'black'
+      });
+    });
+  });
+
+  describe('scheme change listeners', () => {
+    it('should register scheme change listener', () => {
+      const listener = jest.fn();
+      Scheme.addSchemeChangeListener(listener);
+      expect(listener).not.toHaveBeenCalled();
+      Scheme.setScheme('dark');
+      expect(listener).toHaveBeenCalledTimes(1);
+    });
+
+    it('should unregister scheme change listener', () => {
+      const listener = jest.fn();
+      Scheme.addSchemeChangeListener(listener);
+      Scheme.setScheme('dark');
+      expect(listener).toHaveBeenCalledTimes(1);
+      Scheme.removeSchemeChangeListener(listener);
+      Scheme.setScheme('light');
+      expect(listener).toHaveBeenCalledTimes(1);
```

Comment:
We expect the count of calls to stay 1 (instead of 2) 
`toHaveBeenCalledTimes(x);` checks the total calls of a function during this test iteration

If you'll comment this line ->  `Scheme.removeChangeListener(listener);`
You'll see the count will be 2